compiler/src/dotty/tools/dotc/core/unpickleScala2/Scala2Unpickler.scala

```diff
@@ -98,10 +98,11 @@ object Scala2Unpickler {
     val tempInfo = new TempClassInfo(denot.owner.thisType, cls, decls, ost)
     denot.info = tempInfo // first rough info to avoid CyclicReferences
     val parents1 = if (parents.isEmpty) defn.ObjectType :: Nil else parents.map(_.dealias)
-    // Add extra parents to the tuple classes from the standard library
+    // Adjust parents of the tuple classes and BoxedUnit from the standard library
+    // If from Scala 2, adjust for tuple classes; if not, it's from Java, and adjust for BoxedUnit
     val normalizedParents =
       if (fromScala2) defn.adjustForTuple(cls, tparams, parents1)
-      else parents1 // We are setting the info of a Java class, so it cannot be one of the tuple classes
+      else defn.adjustForBoxedUnit(cls, parents1)
```

Comment:
This adjustment also needs to be done in the Namer (like adjustForTuple) since we typecheck BoxedUnit.java when compiling the standard library

Reply:
Is this really how it works in Dotty? Because in scala/scala, `BoxedUnit.java` is compiled by Java first, and then its .class file is used when compiling the .scala sources of the stdlib. I don't think `BoxedUnit.java` is ever processed by scalac. Is the case with dotc?

Reply:
It's going to depend on how the project is built. I see the build.sbt of scala/scala sets `compileOrder := CompileOrder.JavaThenScala` but we don't do that in the dotty repo nor in our tests, so I wouldn't rely on it.

Reply:
OK. Done ;)